solr/docker/templates/Dockerfile.official.header.template

```diff
@@ -21,72 +21,62 @@
 
 FROM _REPLACE_BASE_IMAGE_
 
-# TODO: remove things that exist solely for downstream specialization since Dockerfile.local now exists for that
-
 ARG SOLR_VERSION="_REPLACE_SOLR_VERSION_"
 # empty for the full distribution, "-slim" for the slim distribution
 ARG SOLR_DIST="_REPLACE_SOLR_DIST_"
 ARG SOLR_SHA512="_REPLACE_SOLR_TGZ_SHA_"
 ARG SOLR_KEYS="_REPLACE_RELEASE_MANAGER_GPG_FINGERPRINT_"
 
-# If specified, this will override SOLR_DOWNLOAD_SERVER and all ASF mirrors. Typically used downstream for custom builds
-ARG SOLR_DOWNLOAD_URL
-# TODO: That comment isn't strictly true, if SOLR_DOWNLOAD_URL fails, other mirrors will be attempted
-# TODO: see patch in SOLR-15250 for some example ideas on fixing this to be more strict
-
 # Override the default solr download location with a prefered mirror, e.g.:
 #   docker build -t mine --build-arg SOLR_DOWNLOAD_SERVER=https://downloads.apache.org/solr/solr .
-ARG SOLR_DOWNLOAD_SERVER
-
-# These should never be overridden except for the purposes of testing the Dockerfile before release
-ARG SOLR_CLOSER_URL="http://www.apache.org/dyn/closer.lua?action=download&filename=/solr/solr/$SOLR_VERSION/solr-$SOLR_VERSION$SOLR_DIST.tgz"
-ARG SOLR_DIST_URL="https://www.apache.org/dist/solr/solr/$SOLR_VERSION/solr-$SOLR_VERSION$SOLR_DIST.tgz"
-ARG SOLR_ARCHIVE_URL="https://archive.apache.org/dist/solr/solr/$SOLR_VERSION/solr-$SOLR_VERSION$SOLR_DIST.tgz"
+# This server must support downloading at: ${SOLR_DOWNLOAD_SERVER}/${SOLR_VERSION}/solr-${SOLR_VERSION}(-slim).tgz(.asc)
+ARG SOLR_DOWNLOAD_SERVER="https://www.apache.org/dyn/closer.lua?action=download&filename=/solr/solr"
```

Comment:
I've not checked but just to confirm this needs to support older Solr versions that only live on archive.apache.org right? for say building Solr 9.0 docker image? or when 9.3 is removed from https://www.apache.org/dist/solr/solr/

Reply:
It does, and it works for `.asc` files as well:

https://www.apache.org/dyn/closer.lua?action=download&filename=/solr/solr/9.2.0/solr-9.2.0-src.tgz
https://www.apache.org/dyn/closer.lua?action=download&filename=/solr/solr/9.2.0/solr-9.2.0-src.tgz.asc
https://www.apache.org/dyn/closer.lua?action=download&filename=/solr/solr/9.0.0/solr-9.0.0-src.tgz
https://www.apache.org/dyn/closer.lua?action=download&filename=/solr/solr/9.0.0/solr-9.0.0-src.tgz.asc

(linking the sources because they are smaller)